### DISCUSSION TITLE
Deployment on Azure

### DISCUSSION THREAD

**Discussion by jutglar** [FraunhoferIOSB/FROST-Server#2062] [General]
I need to deploy Frost Server on Azure.

I have deployed the following:
1. Frost Server using Azure Container Apps
2. Azure Database for PostgreSQL

According to the Frost Server documentation (https://fraunhoferiosb.github.io/FROST-Server/deployment/postgresql.html), I need to configure a resource in Tomcat & Wildfly and specify its name in the Frost Server configuration.
However, I'm encountering the following issues while setting up the connection between Frost Server and PostgreSQL:
1. Creating a resource in Tomcat & Wildfly for Azure Database for PostgreSQL
2. Configuring the Azure Container App running Frost Server to recognize the name of that resource.

Do you have experience deploying Frost Server on Azure?
Could you provide a step-by-step guide for this setup?

Thank you very much!

**Comment by hylkevds** [comment]
When using containers everything is configured over environment variables, see [persistence-settings](https://fraunhoferiosb.github.io/FROST-Server/settings/settings.html#persistence-settings).
I'll update that page.

**Reply by jutglar** [reply]
Thank you.
You mentioned the persistence.db.url, persistence.db.username, and persistence.db.password settings. I was also wondering if serviceRootUrl needs to be updated from localhost to the container URL.

Does FROST support SSL mode if required? Azure suggests the below connecting via JDBC. Are the above variables used by FROST to create this connection?
 jdbc:postgresql://{server_name}:5432/postgres?user={your_username}&password={your_password}&sslmode=require

**Reply by hylkevds** [reply]
Typically the same ones as are set in the example [docker-compose](https://github.com/FraunhoferIOSB/FROST-Server/blob/v2.x/scripts/docker-compose-separated.yaml) file.

Yes, the serviceRootUrl must be set to be the URL under which FROST is accessed by clients. Otherwise it can't generate correct links.

You can specify the exact connection URL that FROST should use to connect to PostgreSQL. Though you should leave the username and password out of the connection URL, there are separate environment variables for those.
